source/tests/system/nirfmxnr_driver_api_tests.cpp

```diff
@@ -87,21 +123,622 @@ TEST_F(NiRFmxNRDriverApiTests, Init_Close_Succeeds)
+        break;
+      case NIRFMXNR_INT32_DOWNLINK_TEST_MODEL_TM3_2:
+        PDSCHRMSEVMMean[i] = GET_ATTR_F64(session, carrier_string_response.selector_string_out(), NIRFMXNR_ATTRIBUTE_MODACC_RESULTS_PDSCH_16QAM_RMS_EVM_MEAN);
+        break;
+    }
+    mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response[i] = client::mod_acc_fetch_rmsevm_per_subcarrier_mean_trace(stub(), session, carrier_string_response.selector_string_out(), 10.000000);
+    EXPECT_ERROR(NR_SYNC_FAILURE, NR_SYNC_FAILURE_STR, session, mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response[i]);
+    mod_acc_fetch_rmsevm_per_symbol_mean_trace_response[i] = client::mod_acc_fetch_rmsevm_per_symbol_mean_trace(stub(), session, carrier_string_response.selector_string_out(), 10.000000);
+    EXPECT_ERROR(NR_SYNC_FAILURE, NR_SYNC_FAILURE_STR, session, mod_acc_fetch_rmsevm_per_symbol_mean_trace_response[i]);
+  }
+
+  EXPECT_TRUE(isnan(compositeRMSEVMMean[0]));
+  EXPECT_TRUE(isnan(compositeRMSEVMMean[1]));
+  EXPECT_TRUE(isnan(compositePeakEVMMaximum[0]));
+  EXPECT_TRUE(isnan(compositePeakEVMMaximum[1]));
+  EXPECT_EQ(0, compositePeakEVMSlotIndex[0]);
+  EXPECT_EQ(0, compositePeakEVMSlotIndex[1]);
+  EXPECT_EQ(0, compositePeakEVMSymbolIndex[0]);
+  EXPECT_EQ(0, compositePeakEVMSymbolIndex[1]);
+  EXPECT_EQ(0, compositePeakEVMSubcarrierIndex[0]);
+  EXPECT_EQ(0, compositePeakEVMSubcarrierIndex[1]);
+  EXPECT_TRUE(isnan(componentCarrierFrequencyErrorMean[0]));
+  EXPECT_TRUE(isnan(componentCarrierFrequencyErrorMean[1]));
+  EXPECT_TRUE(isnan(componentCarrierIQOriginOffsetMean[0]));
+  EXPECT_TRUE(isnan(componentCarrierIQOriginOffsetMean[1]));
+  EXPECT_TRUE(isnan(componentCarrierIQGainImbalanceMean[0]));
+  EXPECT_TRUE(isnan(componentCarrierIQGainImbalanceMean[1]));
+  EXPECT_TRUE(isnan(componentCarrierQuadratureErrorMean[0]));
+  EXPECT_TRUE(isnan(componentCarrierQuadratureErrorMean[1]));
+  EXPECT_TRUE(isnan(PDSCHRMSEVMMean[0]));
+  EXPECT_TRUE(isnan(PDSCHRMSEVMMean[1]));
+  EXPECT_EQ(0.0, mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response[0].x0());
+  EXPECT_EQ(0.0, mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response[0].dx());
+  EXPECT_EQ(0, mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response[0].rms_evm_per_subcarrier_mean_size());
+  EXPECT_EQ(0, mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response[0].rms_evm_per_subcarrier_mean().size());
+  EXPECT_EQ(0.0, mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response[1].x0());
+  EXPECT_EQ(0.0, mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response[1].dx());
+  EXPECT_EQ(0, mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response[1].rms_evm_per_subcarrier_mean_size());
+  EXPECT_EQ(0, mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response[1].rms_evm_per_subcarrier_mean().size());
+  // EXPECT_EQ(0.0, mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response.rms_evm_per_subcarrier_mean(0));
```

Comment:
Should we just delete the commented out asserts on the first element when the array is empty?

Reply:
Similar to your new WLAN tests, I'm holding out hope that we can figure out the causes for the errors were getting, and fix them.